test/unit/sql/data-types.test.js

```diff
@@ -161,87 +165,237 @@ suite(Support.getTestDialectTeaser('SQL'), function() {
+        default: 'DOUBLE PRECISION UNSIGNED',
+        postgres: 'DOUBLE PRECISION'
       });
 
-      testsql('FLOAT(11, 12).UNSIGNED.ZEROFILL', DataTypes.FLOAT(11, 12).UNSIGNED.ZEROFILL, {
-        default: 'FLOAT(11,12) UNSIGNED ZEROFILL',
-        sqlite: 'FLOAT UNSIGNED ZEROFILL(11,12)'
+      testsql('DOUBLE(11)', DataTypes.DOUBLE(11), {
+        default: 'DOUBLE PRECISION(11)',
+        postgres: 'DOUBLE PRECISION'
       });
 
-      testsql('FLOAT(11, 12).ZEROFILL', DataTypes.FLOAT(11, 12).ZEROFILL, {
-        default: 'FLOAT(11,12) ZEROFILL',
-        sqlite: 'FLOAT ZEROFILL(11,12)'
+      testsql('DOUBLE(11).UNSIGNED', DataTypes.DOUBLE(11).UNSIGNED, {
+        default: 'DOUBLE PRECISION(11) UNSIGNED',
+        sqlite: 'DOUBLE PRECISION UNSIGNED(11)',
+        postgres: 'DOUBLE PRECISION'
       });
 
-      testsql('FLOAT(11, 12).ZEROFILL.UNSIGNED', DataTypes.FLOAT(11, 12).ZEROFILL.UNSIGNED, {
-        default: 'FLOAT(11,12) UNSIGNED ZEROFILL',
-        sqlite: 'FLOAT UNSIGNED ZEROFILL(11,12)'
+      testsql('DOUBLE(11).UNSIGNED.ZEROFILL', DataTypes.DOUBLE(11).UNSIGNED.ZEROFILL, {
+        default: 'DOUBLE PRECISION(11) UNSIGNED ZEROFILL',
+        sqlite: 'DOUBLE PRECISION UNSIGNED ZEROFILL(11)',
+        postgres: 'DOUBLE PRECISION'
       });
 
+      testsql('DOUBLE(11).ZEROFILL', DataTypes.DOUBLE(11).ZEROFILL, {
+        default: 'DOUBLE PRECISION(11) ZEROFILL',
+        sqlite: 'DOUBLE PRECISION ZEROFILL(11)',
+        postgres: 'DOUBLE PRECISION'
+      });
+
+      testsql('DOUBLE(11).ZEROFILL.UNSIGNED', DataTypes.DOUBLE(11).ZEROFILL.UNSIGNED, {
+        default: 'DOUBLE PRECISION(11) UNSIGNED ZEROFILL',
+        sqlite: 'DOUBLE PRECISION UNSIGNED ZEROFILL(11)',
+        postgres: 'DOUBLE PRECISION'
+      });
+
+      testsql('DOUBLE(11, 12)', DataTypes.DOUBLE(11, 12), {
+        default: 'DOUBLE PRECISION(11,12)',
+        postgres: 'DOUBLE PRECISION'
+      });
+
+      testsql('DOUBLE(11, 12).UNSIGNED', DataTypes.DOUBLE(11, 12).UNSIGNED, {
+        default: 'DOUBLE PRECISION(11,12) UNSIGNED',
+        sqlite: 'DOUBLE PRECISION UNSIGNED(11,12)',
+        postgres: 'DOUBLE PRECISION'
+      });
+
+      testsql('DOUBLE(11, 12).UNSIGNED.ZEROFILL', DataTypes.DOUBLE(11, 12).UNSIGNED.ZEROFILL, {
+        default: 'DOUBLE PRECISION(11,12) UNSIGNED ZEROFILL',
+        sqlite: 'DOUBLE PRECISION UNSIGNED ZEROFILL(11,12)',
+        postgres: 'DOUBLE PRECISION'
+      });
+
+      testsql('DOUBLE(11, 12).ZEROFILL', DataTypes.DOUBLE(11, 12).ZEROFILL, {
+        default: 'DOUBLE PRECISION(11,12) ZEROFILL',
+        sqlite: 'DOUBLE PRECISION ZEROFILL(11,12)',
+        postgres: 'DOUBLE PRECISION'
+      });
+
+      testsql('DOUBLE(11, 12).ZEROFILL.UNSIGNED', DataTypes.DOUBLE(11, 12).ZEROFILL.UNSIGNED, {
+        default: 'DOUBLE PRECISION(11,12) UNSIGNED ZEROFILL',
+        sqlite: 'DOUBLE PRECISION UNSIGNED ZEROFILL(11,12)',
+        postgres: 'DOUBLE PRECISION'
+      });
     });
 
+    if (current.dialect.name !== 'postgres') {
```

Comment:
Actually just kill the IF entirely and provide what we actually do, we do some sort of transformation at the moment.